overlord/snapshotstate/snapshotstate.go

```diff
@@ -0,0 +1,268 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snapshotstate
+
+import (
+	"fmt"
+	"sort"
+
+	"golang.org/x/net/context"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/overlord/snapshotstate/backend"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/strutil"
+)
+
+var (
+	snapstateAll                     = snapstate.All
+	snapstateCheckChangeConflictMany = snapstate.CheckChangeConflictMany
+	backendIter                      = backend.Iter
+)
+
+type snapshotState struct {
+	LastSetID uint64 `json:"last-set-id"`
+}
+
+func newSnapshotSetID(st *state.State) (uint64, error) {
+	var shotState snapshotState
+
+	err := st.Get("snapshots", &shotState)
```

Comment:
It depends. Maybe? Dunno.

Reply:
to be clear, I'm happy with things as they stand: if the worst case is that it ends with snapshots.snapshots, then it's not too bad. Otherwise let's sync with @niemeyer and you.

Reply:
Alright, how about going with the more familiar "last-snapshot-set-id"? We have other similar cases in the top-level already.